pyqtorch/core/operation.py

```diff
@@ -251,6 +271,25 @@ def H(state: torch.Tensor, qubits: ArrayLike, N_qubits: int) -> torch.Tensor:
     return _apply_gate(state, mat, qubits, N_qubits)
 
 
+def ControlledOperationGate(state: torch.Tensor, qubits: ArrayLike, N_qubits: int, operation_matrix: torch.Tensor) -> torch.Tensor:
+    """Generalized Controlled Rotation gate with two-qubits support
+
+    Args:
+        state (torch.Tensor): the input quantum state, of shape `(N_0, N_1,..., N_N, batch_size)`
+        qubits (ArrayLike): list of qubit indices where the gate will operate
+        N_qubits (int): the number of qubits in the system
+        operation_type (str): the type of operation which should be performed (NOT,RX,RY,RZ)
```

Comment:
`operation_type` does not seem to appear among the arguments. Are you planning to pass only the operation matrix? Also the `maybe_theta` argument is not there anymore.

Reply:
reworked the documentation to be accurate